x-django-service: &django-service
  user: "${UID}:${GID}"
  build: .
  volumes:
    - .:/code
  depends_on:
    db:
      condition: service_healthy

x-django-eng: &django-env
  PYTHONUNBUFFERED: 1
  DATABASE_URL: "postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db/${POSTGRES_DB}"
  DJANGO_SECRET_KEY: "${DJANGO_SECRET_KEY}"

services:
  db:
    image: postgres:14
    environment:
      POSTGRES_USER: "${POSTGRES_USER}"
      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
      POSTGRES_DB: "${POSTGRES_DB}"
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}" ]
      interval: 5s
      timeout: 5s
      retries: 5
    ports:
      - "5434:5432"

  web:
    <<: *django-service
    command: ${RUNSERVER_CMD}
    extra_hosts:
      - "host.docker.internal:host-gateway"
    ports:
      - "${DEV_PORT}:8000"
    environment:
      <<: *django-env

  worker:
    <<: *django-service
    command: python -m processing
    restart: always
    extra_hosts:
      - "host.docker.internal:host-gateway"
    environment:
      <<: *django-env
      DJANGO_SETTINGS_MODULE: "timings.settings"

volumes:
  postgres_data:
    driver: local
